kustomize/data-plane/shard/operator-csv-patch.json

```diff
@@ -0,0 +1,15 @@
+[
+  { "op":  "add",
```

Comment:
@winklerm Why do we need this patch?

Reply:
The patch was needed in order to set parameters for the shard operator - since we use subscription to install it, we cannot specify the parameters directly in the deployment. The operator needs to be installed, which creates Operator CSV with a deployment template, which can be patched then.

When I tried to learn about possible approach to this, I was suggested also to provide the configuration for the operator as a custom resource, that way it could work even when the operator is updated to a newer version (as opposed to setting the operator parameters in the CSV deployment template which would probably get overwritten with an operator update). However, that would mean additional work, not sure if that would make sense fro your perspective, though?

Right now I have a feeling that this could be replaced by configuring an addon for the shard-operator (afaik, addons support parameters), not sure if that could be used outside of the managed environment?

Reply:
@winklerm Thank you for the extra context.

I think if this gets you up and running, let's continue with it. But I think we should probably fix it longer-term. The order of fix preference I would say is:

1. Make the CVP pipeline use OSD and AddOn model of deployment. This would have the additional benefit of aligning the CVP pipeline more closely with our `dev`, `stage` and `prod` environments.
2. We could do something with the Quarkus Kubernetes Config extension https://quarkus.io/guides/kubernetes-config to ensure that the Operator always reads a particular `ConfigMap` or `Secret` from the `namespace` in which it is installed.

WDYT?